algopy_testing/src/algopy_testing/context.py

```diff
@@ -605,51 +693,92 @@ def any_app_call_txn(  # noqa: PLR0913
         }
-        new_txn = algopy.gtxn.ApplicationCallTransaction(group_index)
+        new_txn = algopy.gtxn.ApplicationCallTransaction(NULL_GTXN_GROUP_INDEX)
 
         for key, value in {**kwargs, **callable_params}.items():
             setattr(new_txn, key, value)
 
         return new_txn
 
     def any_axfer_txn(
-        self, group_index: int, **kwargs: Unpack[AssetTransferFields]
+        self, **kwargs: Unpack[AssetTransferFields]
     ) -> algopy.gtxn.AssetTransferTransaction:
         """
         Generate a new asset transfer transaction with specified fields.
 
         Args:
-            group_index (int): Position index in the transaction group.
             **kwargs (Unpack[AssetTransferFields]): Fields to be set in the transaction.
 
         Returns:
             algopy.gtxn.AssetTransferTransaction: The newly generated asset transfer transaction.
         """
         import algopy.gtxn
 
-        new_txn = algopy.gtxn.AssetTransferTransaction(group_index)
+        new_txn = algopy.gtxn.AssetTransferTransaction(NULL_GTXN_GROUP_INDEX)
 
         for key, value in kwargs.items():
             setattr(new_txn, key, value)
 
         return new_txn
 
-    def any_pay_txn(
-        self, group_index: int, **kwargs: Unpack[PaymentFields]
-    ) -> algopy.gtxn.PaymentTransaction:
+    def any_pay_txn(self, **kwargs: Unpack[PaymentFields]) -> algopy.gtxn.PaymentTransaction:
         """
         Generate a new payment transaction with specified fields.
 
         Args:
-            group_index (int): Position index in the transaction group.
             **kwargs (Unpack[PaymentFields]): Fields to be set in the transaction.
 
         Returns:
             algopy.gtxn.PaymentTransaction: The newly generated payment transaction.
         """
         import algopy.gtxn
 
-        new_txn = algopy.gtxn.PaymentTransaction(group_index)
+        new_txn = algopy.gtxn.PaymentTransaction(NULL_GTXN_GROUP_INDEX)
+
+        for key, value in kwargs.items():
+            setattr(new_txn, key, value)
+
+        return new_txn
+
+    def any_acfg_txn(
+        self, **kwargs: Unpack[AssetConfigFields]
+    ) -> algopy.gtxn.AssetConfigTransaction:
+        """
+        Generate a new ACFG transaction with specified fields.
+        """
+        import algopy.gtxn
+
+        new_txn = algopy.gtxn.AssetConfigTransaction(NULL_GTXN_GROUP_INDEX)
+
+        for key, value in kwargs.items():
+            setattr(new_txn, key, value)
+
+        return new_txn
+
+    def any_keyreg_transaction(
+        self, **kwargs: Unpack[KeyRegistrationFields]
+    ) -> algopy.gtxn.KeyRegistrationTransaction:
+        """
+        Generate a new key registration transaction with specified fields.
+        """
+        import algopy.gtxn
+
+        new_txn = algopy.gtxn.KeyRegistrationTransaction(NULL_GTXN_GROUP_INDEX)
+
+        for key, value in kwargs.items():
+            setattr(new_txn, key, value)
+
+        return new_txn
+
+    def any_afrz_transaction(
```

Comment:
Should we use `asset_freeze` rather than `afrz`, I assume we use full names in algorand python?

Reply:
Same for the others, e.g. acfg etc.

Reply:
@robdmoore the terms are not gonna be well knows for newcomers I agree, on the other hands they are fairly well understood by existing devs. The only downside with full names is more verbosity in invoking these methods. But other than that can rename them to full names

Reply:
We want puya to be approachable for new devs. Verbosity doesn't matter vs the clarity in my opinion